java/memory/src/main/java/org/apache/arrow/memory/util/hash/SimpleHasher.java

```diff
@@ -0,0 +1,126 @@
+import static io.netty.util.internal.PlatformDependent.getInt;
+import static io.netty.util.internal.PlatformDependent.getLong;
+
+import org.apache.arrow.memory.util.ArrowBufPointer;
+
+import io.netty.buffer.ArrowBuf;
+
+/**
+ * A simple hasher that calculates the hash code of integers as is,
+ * and does not perform any finalization. So the computation is extremely
+ * efficient.
+ * <p>
+ *   This algorithm only provides the most basic semantics for the hash code. That is,
+ *   if two objects are equal, they must have equal hash code. However, the quality of the
+ *   produced hash code may not be good. In other words, the generated hash codes are
+ *   far from being uniformly distributed in the universe.
+ * </p>
+ * <p>
+ *   Therefore, this algorithm is suitable only for scenarios where the most basic semantics
+ *   of the hash code is required (e.g. in scenarios that require fast and proactive data pruning)
+ * </p>
+ * <p>
+ *   An object of this class is stateless, so it can be shared between threads.
+ * </p>
+ */
+public class SimpleHasher implements ArrowBufHasher {
+
+  public static SimpleHasher INSTANCE = new SimpleHasher();
+
+  protected SimpleHasher() {
+  }
+
+  /**
+   * Calculates the hash code for a memory region.
+   * @param address start address of the memory region.
+   * @param length length of the memory region.
+   * @return the hash code.
+   */
+  public int hashCode(long address, int length) {
+    int hashValue = 0;
+    int index = 0;
+    while (index + 8 <= length) {
+      long longValue = getLong(address + index);
+      if (!ArrowBufPointer.LITTLE_ENDIAN) {
+        // assume the buffer is in little endian
+        longValue = Long.reverseBytes(longValue);
+      }
+      int longHash = getLongHashCode(longValue);
+      hashValue = combineHashCode(hashValue, longHash);
+      index += 8;
+    }
+
+    while (index + 4 <= length) {
+      int intValue = getInt(address + index);
+      if (!ArrowBufPointer.LITTLE_ENDIAN) {
+        intValue = Integer.reverseBytes(intValue);
+      }
+      int intHash = intValue;
+      hashValue = combineHashCode(hashValue, intHash);
+      index += 4;
+    }
+
+    while (index < length) {
+      byte byteValue = getByte(address + index);
+      int byteHash = byteValue;
+      hashValue = combineHashCode(hashValue, byteHash);
+      index += 1;
+    }
+
+    return finalizeHashCode(hashValue);
+  }
+
+  /**
+   * Calculates the hash code for a memory region.
+   * @param buf the buffer for the memory region.
+   * @param offset offset within the buffer for the memory region.
+   * @param length length of the memory region.
+   * @return the hash code.
+   */
+  public int hashCode(ArrowBuf buf, int offset, int length) {
+    buf.checkBytes(offset, offset + length);
+    return hashCode(buf.memoryAddress() + offset, length);
+  }
+
+  protected final int combineHashCode(int currentHashCode, int newHashCode) {
```

Comment:
not required any longer ?

Reply:
I am not sure I understand your question.
You mean the method is no longer required, or the protected modifier is no longer required?

Reply:
sorry, I was confused with the ArrowBufHasher.